packages/@react-spectrum/numberfield/docs/NumberField.mdx

```diff
@@ -0,0 +1,218 @@
+the `id` of the labeling element instead.
+
+### Internationalization
+
+#### Label
+In order to internationalize a NumberField, a localized string should be passed to the `label` or `aria-label` prop.
+When the `necessityIndicator` prop is set to `"label"`, a localized string will be provided for `"(required)"` or `"(optional)"` automatically.
+
+#### Value
+The NumberField value can be formatted by using the `formatOptions` prop.
+`formatOptions` is compatible with the option parameter of [Intl.NumberFormat](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/NumberFormat) and is applied based on the current locale.
+
+```tsx example
+<Flex gap="size-150" wrap>
+  <NumberField
+    label="Currency"
+    formatOptions={{style: 'currency', currency: 'JPY'}}
+    defaultValue={60} />
+  <NumberField
+    label="Percent"
+    maxValue={1}
+    step={0.001}
+    formatOptions={{style: 'percent', minimumFractionDigits: 1}}
+    defaultValue={0.891} />
+</Flex>
+```
+
+NumberField also supports three numeral systems, Latin digits, Arabic-Indic digits, and Hanidecimal. Regardless of the locale, these three
+can be parsed from typed input. In addition, the NumberField will automatically choose the starting numeral system based on the default for the locale.
+
+
+## Events
+
+NumberField accepts an `onChange` prop which is triggered whenever the value is committed by the user. This happens on blur of the field
+or on interaction with the stepper functionality, arrow keys or stepper buttons.
+For a full list of supported events, see the [Props](#props) table below.
+
+The example below uses `onChange` to update a separate `pre` element with the same values entered into the NumberField.
+```tsx example
+function Example() {
+  let [value, setValue] = React.useState();
+
+  return (
+    <Flex direction="column" gap="size-150">
+      <NumberField
+        onChange={setValue}
+        label="Number of cookies to buy" />
+      <pre>How many cookies you are ordering: {isNaN(value) ? 0 : value}</pre>
+    </Flex>
+  );
+}
+```
+
+## Validation
+
+NumberFields can display a validation state to communicate to the user whether the current value is valid or invalid.
+Implement your own validation logic in your app and pass either `"valid"` or `"invalid"` to the NumberField via the `validationState` prop.
+
+The example below illustrates how one would validate if the user has entered a [FizzBuzz](https://en.wikipedia.org/wiki/Fizz_buzz) number into the NumberField.
+```tsx example
+function Example() {
+  let [value, setValue] = React.useState(15);
+  let isValid = React.useMemo(() => value % 3 === 0 && value % 5 === 0, [value]);
+
+  return (
+    <NumberField
+      validationState={isValid ? 'valid' : 'invalid'}
+      value={value}
+      onChange={setValue}
+      label="FizzBuzz values only"
+    />
+  );
+}
+```
+
+## Props
+
+<PropTable component={docs.exports.NumberField} links={docs.links} />
+
+## Visual options
```

Comment:
I feel like we should discuss the display of this in large theme for mobile since it is so different.

Reply:
What do we need to say? We don't really talk about what the Desktop version looks like, we just assume they are coming from Desktop. If they're coming from looking at the docs on Mobile, should we tell them what it'll look like on Desktop?

Reply:
I took a look at ComboBox docs and we don't talk about the mobile view there, so this is fine.

My thought process was that we made an accessibility change to ComboBox when a tray is used that is visually different and it seems like that should be explained to someone reading the docs. And the same with NumberField and different step buttons. The other case I can think of right now is tray, which we document in DialogTrigger saying we created this for mobile. The difference is tray is a mode we can force with a prop.